core/src/main/java/nl/nn/adapterframework/jta/StatusRecordingTransactionManager.java

```diff
@@ -0,0 +1,147 @@
+/*
+   Copyright 2021 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.jta;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import javax.transaction.TransactionManager;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.transaction.TransactionSystemException;
+import org.springframework.transaction.jta.JtaTransactionManager;
+import org.springframework.util.StreamUtils;
+
+import lombok.Getter;
+import lombok.Setter;
+import nl.nn.adapterframework.util.LogUtil;
+import nl.nn.credentialprovider.util.Misc;
+
+/**
+ * JtaTransactionManager-wrapper that enables to recover transaction logs produced by another instance.
+ *  
+ * @author Gerrit van Brakel
+ *
+ */
+public abstract class StatusRecordingTransactionManager extends JtaTransactionManager implements DisposableBean {
+	protected Logger log = LogUtil.getLogger(this);
+
+	private static final long serialVersionUID = 1L;
+	private static final int TMUID_MAX_LENGTH=51;
+	
+	private @Getter @Setter String statusFile;
+	private @Getter @Setter String tmuidFile;
+	private @Getter @Setter String tmuid;
+
+	private volatile TransactionManager transactionManager;
+
+	protected enum Status {
+		INITIALIZING,
+		ACTIVE,
+		PENDING,
+		COMPLETED;
+	}
+	
+	/**
+	 * shut down the transaction manager, attempting to complete all running transactions
+	 * @return true if any transactions are pending, that need to be recovered later. 
+	 */
+	protected abstract TransactionManager createTransactionManager() throws TransactionSystemException;
+	protected abstract boolean shutdownTransactionManager() throws TransactionSystemException;
+
+	@Override
+	protected TransactionManager retrieveTransactionManager() throws TransactionSystemException {
```

Comment:
ik zou zeggen getTransactionManager

Reply:
N.B. retrieveTransactionManager() is called once from JtaTransactionManager.initUserTransactionAndTransactionManager(),